src/shims/env.rs

```diff
@@ -150,4 +154,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn update_environ(&mut self) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        // Collect all the pointers to each variable in a vector.
+        let mut vars: Vec<Scalar<Tag>> = this.machine.env_vars.map.values().map(|&ptr| ptr.into()).collect();
+        // Add the trailing null pointer.
+        vars.push(Scalar::from_int(0, this.pointer_size()));
+        // Make an array with all these pointers inside Miri.
+        let tcx = this.tcx;
+        let environ_layout =
+            this.layout_of(tcx.mk_array(tcx.mk_imm_ptr(tcx.types.u8), vars.len() as u64))?;
+        let environ_place = this.allocate(environ_layout, MiriMemoryKind::Machine.into());
```

Comment:
Please don't call this `environ` too, that's awfully confusing to have two *different* allocations/pointers with the same name.^^

Maybe call it `environ_list`?

Reply:
Or maybe call it `vars`, matching your `Vec` :)

Reply:
hahaha ok. I'll rename them to `vars_*`